sulong/projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/intrinsics/multithreading/CConstants.java

```diff
@@ -0,0 +1,139 @@
+package com.oracle.truffle.llvm.nodes.intrinsics.multithreading;
+
+import com.oracle.truffle.api.RootCallTarget;
+import com.oracle.truffle.llvm.runtime.LLVMContext;
+import com.oracle.truffle.llvm.runtime.LLVMLanguage;
+import com.oracle.truffle.llvm.runtime.memory.LLVMStack;
+
+public class CConstants {
+    private static LLVMContext ctx;
```

Comment:
ok, the fields and methods are now non-static and a final instance of that class is in the context now, which has a final reference to that context.
also renamed to "PThreadCConstants" and moved to package "com.oracle.truffle.llvm.runtime.util" to avoid a circular dependency (warning).